src/tabs/Tabs.tsx

```diff
@@ -135,9 +140,11 @@ export function Tabs({
   children = Children.toArray(children).filter(child => child);
   const tabs = parseTabList(children);
   // Initialize the selected tab to the first non-hidden tab
-  const [selectedIndex, setSelectedIndex] = useState<number>(
+  const [selectedStateIndex, setSelectedIndex] = useState<number>(
     tabs.findIndex(tab => !tab.hidden)
   );
+  const selectedIndex =
+    typeof selectedTabIndex == "number" ? selectedTabIndex : selectedStateIndex;
```

Comment:
Not blocking but you could encapsulate all this in a hook like `useControlledState`

https://github.com/reach/reach-ui/blob/43f450db7bcb25a743121fe31355f2294065a049/packages/utils/src/use-controlled-state.ts#L14

Reply:
done!